packages/core/src/lib/modal.svelte

```diff
@@ -26,81 +26,84 @@ Creates a modal overlay.
 
 <script lang="ts">
 import cx from 'classnames';
-import { createEventDispatcher, onMount } from 'svelte';
+import { createEventDispatcher } from 'svelte';
 import IconButton from './button/icon-button.svelte';
+import { clickOutside } from '$lib';
 
 /** Whether the modal is open. */
 export let open = false;
+
+$: if (typeof document !== 'undefined') {
+  document.body.classList.toggle('overflow-hidden', open);
+}
+
 /** The variant of the modal. */
 export let variant: 'small' | '' = '';
 
 const dispatch = createEventDispatcher<{
   /** Fires when the modal is requested to close. */
-  close: boolean;
+  close: void;
 }>();
 
 const handleCloseModal = () => {
-  dispatch('close', true);
-};
-
-const handleBackgroundClick = (event: MouseEvent) => {
-  if (event.target === event.currentTarget) {
-    handleCloseModal();
+  if (!open) {
+    return;
   }
+  dispatch('close');
```

Comment:
[Nit] Generally I'm a fan of early returns in complex chains, but given we've got a single condition and a single action here, I think this is easier to read if we flip the condition:

```suggestion
  if (!open) {
    dispatch('close');
  }
  ```